src/Core/Manifest/VersionProvider.php

```diff
@@ -62,6 +62,23 @@ public function getVersion()
         return implode(', ', $ret);
     }
 
+    /**
+     * Get the version of a specific module
+     * Will strip out the leading "module name" from the returned value
+     *
+     * @param string $module - e.g. silverstripe/framework
+     * @return string - e.g. 4.10
+     */
+    public function getModuleVersion(string $module): string
+    {
+        $previousModules = $this->getModules();
+        Config::modify()->set(self::class, 'modules', [$module => 'MODULE_NAME']);
+        $version = $this->getVersion();
+        $version = str_replace('MODULE_NAME: ', '', $version);
+        Config::modify()->set(self::class, 'modules', $previousModules);
```

Comment:
This seems a bit of a strange way to do it - config modifications are generally something we want to cut down on as they’re quite expensive calls IIRC - can’t we do a preg_replace or preg_split on `:`?

Reply:
Not really.  If we run $this->getVersion() we only get the 'selected' version e.g. `"Core Recipe: 7.7.7"` .. where 'Core Recipe' is the 'name' for silverstripe/cms

Also, what happens when the module you want isn't in the existing config? What happens when you request silverstripe/asset-admin?

Reply:
How about this?

```php
public function getModuleVersion(string $module): string
{
    $lockedModules = $this->getModuleVersionFromComposer([$module]);
    return $lockedModules[$module] ?? '';
}
```

Reply:
Yeah ... I'm with @kinglozzer, I really don't understand the purpose of the Config calls here. Is this meant to be some sort of caching mechanism?

Reply:
@kinglozzer yup that's a good spot, I've redone the function to use `getModuleVersionFromComposer()` and have put caching around it